### PR TITLE
Refactoring Topics reducer into redux toolkit

### PR DESCRIPTION
**Pull request by rAzizbekyan**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [ ] Manually (please, describe, if necessary)
- [ ] Unit checks
- [x] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/redux/reducers/topics/topicsSlice.ts [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+  TopicsState,
+  TopicName,
+  TopicFormData,
+  TopicFormFormattedParams,
+  TopicFormDataRaw,
+  ClusterName,
+} from 'redux/interfaces';
+import { BASE_PARAMS } from 'lib/constants';
+import { getResponse } from 'lib/errorHandling';
+import { showSuccessAlert } from 'redux/reducers/alerts/alertsSlice';
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/redux/reducers/topics/topicsSlice.ts:34]: i think here we have to remove this never used import

File: kafka-ui-react-app/src/components/Topics/New/New.tsx [hunk 2]
```diff
@@ -3,17 +3,12 @@ import { ClusterName, TopicFormData, FailurePayload } from 'redux/interfaces';
...
 import { clusterTopicPath } from 'lib/paths';
 import TopicForm from 'components/Topics/shared/Form/TopicForm';
-import {
-  formatTopicCreation,
-  topicsApiClient,
-  createTopicAction,
-} from 'redux/actions';
-import { useDispatch } from 'react-redux';
-import { getResponse } from 'lib/errorHandling';
+import { createTopic } from 'redux/reducers/topics/topicsSlice';
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/New/New.tsx:6]: FailurePayload remove cause it is not used

**Review by Mgrdich — COMMENTED** [review]
there are several pipeline errors that indicates , an unused variables those should be fixed.

**Review by sasunprov — APPROVED** [review]

File: kafka-ui-react-app/src/components/Topics/List/__tests__/List.spec.tsx [hunk 3]
```diff
@@ -290,10 +290,14 @@ describe('List', () => {
...
 
       expect(mockFn).toBeCalledTimes(1);
-      expect(mockFn).toBeCalledWith('local', [
-        externalTopicPayload.name,
-        'external.topic2',
-      ]);
+      // expect(mockFn).toBeCalledWith('local', [
+      //   externalTopicPayload.name,
+      //   'external.topic2',
+      // ]);
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/List/__tests__/List.spec.tsx:296]: pls remove unused code

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 4]
```diff
@@ -119,32 +121,58 @@ describe('New', () => {
...
   });
 
-  it('submits valid form that result in an error', async () => {
-    const createTopicAPIPathMock = fetchMock.postOnce(
-      createTopicAPIPath,
-      { throws: new Error('Something went wrong') },
-      {
-        body: createTopicPayload,
-      }
+  it('not to redirect if request is not fulfilled', async () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:129]:
```suggestion
  it('does not redirect page  when request is not fulfilled', async () => {
```

**Review by lazzy-panda — APPROVED** [review]

**Review by ssiradeghyan — APPROVED** [review]

**Pull request merged**